If we want to stop distributing *ph, I need a way to get at TIOCGWINSZ that's
not too hacky.

millert@ suggested Term::ReadKey, which looks indeed like a good choice.

Here's the glue + distrib from cpan. Comments welcome.

--- ./MANIFEST.orig	Thu Jul 17 15:49:09 2025
+++ ./MANIFEST	Thu Jul 17 15:49:16 2025
@@ -2318,6 +2318,17 @@ cpan/Term-ANSIColor/t/module/true-color.t
 cpan/Term-ANSIColor/t/taint/basic.t
 cpan/Term-Cap/Cap.pm			Perl module supporting termcap usage
 cpan/Term-Cap/test.pl			See if Term::Cap works
+cpan/Term-ReadKey/Changes			Term::ReadKey
+cpan/Term-ReadKey/Configure.pm			Term::ReadKey
+cpan/Term-ReadKey/example/test.pl		Term::ReadKey
+cpan/Term-ReadKey/genchars.pl			Term::ReadKey
+cpan/Term-ReadKey/Makefile.PL			Term::ReadKey
+cpan/Term-ReadKey/ppport.h			Term::ReadKey
+cpan/Term-ReadKey/ReadKey.xs			Term::ReadKey
+cpan/Term-ReadKey/ReadKey.pm.PL			Term::ReadKey
+cpan/Term-ReadKey/README			Term::ReadKey
+cpan/Term-ReadKey/t/01_basic.t			Term::ReadKey
+cpan/Term-ReadKey/t/02_terminal_functions.t	Term::ReadKey
 cpan/Term-Table/lib/Term/Table.pm		Term-Table
 cpan/Term-Table/lib/Term/Table/Cell.pm		Term-Table
 cpan/Term-Table/lib/Term/Table/CellStack.pm	Term-Table
--- ./Makefile.SH.orig	Thu Jul 17 15:49:08 2025
+++ ./Makefile.SH	Thu Jul 17 15:49:12 2025
@@ -217,6 +217,7 @@ extra_dep='
 cpan/Pod-Simple/pm_to_blib: dist/if/pm_to_blib
 ext/Pod-Functions/pm_to_blib: cpan/Pod-Simple/pm_to_blib cpan/Pod-Escapes/pm_to_blib pod/perlfunc.pod
 cpan/IO-Compress/pm_to_blib: dist/lib/pm_to_blib
+lib/auto/Term/ReadKey/ReadKey.so: lib/auto/Cwd/Cwd.so
 '
 for f in $dynamic_ext; do
     : the dependency named here will never exist
--- ./regen/lib_cleanup.pl.orig	Tue Jun 24 08:23:21 2025
+++ ./regen/lib_cleanup.pl	Thu Jul 17 15:49:12 2025
@@ -74,6 +74,12 @@ foreach my $file (@ext) {
                 $package = $1;
                 last;
             }
+            elsif (/^\s*package\s*$/) {
+                # If they're hiding their package name, we ignore them
+                ++$ignore{"/$path"};
+                $package='';
+                last;
+            }
         }
         close $fh
             or die "Can't close $file: $!";
